pandas/_libs/src/parser/tokenizer.c

```diff
@@ -1834,201 +1839,196 @@ int uint64_conflict(uint_state *self) {
   return self->seen_uint && (self->seen_sint || self->seen_null);
 }
 
+/**
+ * @brief Check if the character in the pointer indicates a number.
+ * It expects that you consumed all leading whitespace.
+ *
+ * @param p_item Pointer to verify
+ * @return Non-zero integer indicating that has a digit 0 otherwise.
+ */
+static inline bool has_digit_int(const char *str) {
+  if (!str || *str == '\0') {
+    return false;
+  }
+
+  switch (*str) {
+  case '0':
+  case '1':
+  case '2':
+  case '3':
+  case '4':
+  case '5':
+  case '6':
+  case '7':
+  case '8':
+  case '9':
+    return true;
+  case '+':
+  case '-':
+    return str[1] != '\0' && isdigit_ascii(str[1]);
+  default:
+    return false;
+  }
+}
+
+static inline bool has_only_spaces(const char *str) {
+  while (*str != '\0' && isspace_ascii(*str)) {
+    str++;
+  }
+  return *str == '\0';
+}
+
+/* Copy a string without `char_to_remove` into `output`.
+ */
+static int copy_string_without_char(char output[PROCESSED_WORD_CAPACITY],
+                                    const char *str, size_t str_len,
+                                    char char_to_remove) {
+  const char *left = str;
+  const char *right;
+  const char *end_ptr = str + str_len;
+  size_t bytes_written = 0;
+
+  while (left < end_ptr) {
+    size_t remaining_bytes_to_read = end_ptr - left;
+    right = memchr(left, char_to_remove, remaining_bytes_to_read);
+
+    if (!right) {
+      // If it doesn't find the char to remove, just copy until EOS.
+      right = end_ptr;
+    }
+
+    size_t chunk_size = right - left;
+
+    // check if we have enough space, including the null terminator.
```

Comment:
I think the comments here and beyond are all superfluous, i.e. they add no value on top of what the code itself is showing